pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -275,22 +278,207 @@ func getErrorAsStatus(err error) *framework.Status {
+func (pl *VolumeZone) isSchedulableAfterPersistentVolumeClaimAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, addedPVC, err := util.As[*v1.PersistentVolumeClaim](nil, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeClaimAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	pvcNames := pl.getPersistentVolumeClaimClassNameFromPod(logger, pod)
+	for _, pvcName := range pvcNames {
+		if (addedPVC.Name == pvcName) && (pod.Namespace == addedPVC.Namespace) {
+			logger.V(4).Info("PVC was created, and matches the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(addedPVC))
+			return framework.QueueAfterBackoff
+		}
+	}
+
+	logger.V(4).Info("PVC was created, but it doesn't match the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(addedPVC))
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterPersistentVolumeChange is invoked whenever a PersistentVolume added or updated. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeZone) isSchedulableAfterPersistentVolumeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	originalPV, modifiedPV, err := util.As[*v1.PersistentVolume](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+	originalPVTopologies := make([]pvTopology, 0)
+	if originalPV != nil {
+		originalPVTopologies, _ = pl.getPVTopologiesFromPV(logger, pod, originalPV)
+	}
+	modifiedPVTopologies, isPodBoundToPV := pl.getPVTopologiesFromPV(logger, pod, modifiedPV)
+	if isPodBoundToPV && !reflect.DeepEqual(originalPVTopologies, modifiedPVTopologies) {
+		logger.V(4).Info("PV was created or updated, and be able to schedule the pod", "pod", klog.KObj(pod), "PV", klog.KObj(modifiedPV))
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("PV was created or updated, but it doesn't make this pod schedulable", "pod", klog.KObj(pod), "PV", klog.KObj(modifiedPV))
+	return framework.QueueSkip
+}
+
+// isPVBoundToPVC checks whether the PV is bound to pod's PVC.
+func (pl *VolumeZone) getPVTopologiesFromPV(logger klog.Logger, pod *v1.Pod, pv *v1.PersistentVolume) ([]pvTopology, bool) {
+	isPodBoundToPV := false
+	podPVTopologies := make([]pvTopology, 0)
+
+	pvcNames := pl.getPersistentVolumeClaimClassNameFromPod(logger, pod)
+	for _, pvcName := range pvcNames {
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
+		if s := getErrorAsStatus(err); !s.IsSuccess() {
+			logger.Error(err, "unexpected error during getting PVC", "Namespace", pod.Namespace, "pvc", pvcName)
+			continue
+		}
+
+		pvName := pvc.Spec.VolumeName
+		if pvName == pv.Name {
+			isPodBoundToPV = true
+			for _, key := range topologyLabels {
```

Comment:
This part is also used in other functions (line 450-line 463). So I'm thinking reuse these codes.
Maybe we can split this function into three function?

1. `IsPvBoundToPod` to check if a pv belongs to a pod.
2. `GetTopologyFromPV` to get topology labels from a pv. This can also be used in `getPvBypod`.
3. `Equals(a []pvTopology, b []pvTopology)`. To check if two label slice is equal, and be called when it's needed.
Just a thought, WDYT

Reply:
Thanks. Fixed it.

About No1 (IsPVBoundToPod), I didn't create this function because this function is called once.
Thanks to your comment, code change became clear. I appreciate it